### PR TITLE
Support for webpack 4

### PR DESCRIPTION
**Pull request by dmitmel**
Related to #255

### CODE REVIEW

File: packages/dev-server/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # @webpack-blocks/dev-server - Changelog
 
+## Next Release
+
+- Removed `NamedModulesPlugin` because webpack 4 automatically adds it in the `development` mode ([#255](https://github.com/andywer/webpack-blocks/issues/255))
```
Comment (vlad-zhukov) [inline packages/dev-server/CHANGELOG.md]: #261

File: packages/dev-server/package.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
   "repository": "andywer/webpack-blocks",
   "bugs": "https://github.com/andywer/webpack-blocks/issues",
   "dependencies": {
-    "webpack-dev-server": "^2.9.1"
+    "webpack-dev-server": "^2.11.2"
```
Comment (vlad-zhukov) [inline packages/dev-server/package.json]: `webpack-dev-server` 3+?

File: packages/dev-server/package.json [hunk 3]
```diff
@@ -19,7 +19,7 @@
...
   },
   "peerDependencies": {
     "webpack": "2.x || 3.x"
```
Comment (vlad-zhukov) [inline packages/dev-server/package.json]: It's going to be `v4` only.

Comment (vlad-zhukov) [reply packages/dev-server/package.json]: Here and in other places

File: packages/dev-server/CHANGELOG.md (hunk 1 above)
Comment (dmitmel) [reply packages/dev-server/CHANGELOG.md]: Fixed

File: packages/dev-server/package.json (hunk 2 above)
Comment (dmitmel) [reply packages/dev-server/package.json]: Fixed

File: packages/dev-server/package.json (hunk 3 above)
Comment (dmitmel) [reply packages/dev-server/package.json]: Fixed

**Comment by jeremy-clearlabs** [comment]
@dmitmel Sent you a PR on your fork for updating your pull request that resolves the conflicts.

**Comment by vlad-zhukov** [comment]
There are too many changes for a single PR. It's very difficult to review it and I don't want to miss something important out. I would like to migrate to `yarn` first, then it will be much easier to work on this PR as well as others.

**Comment by vlad-zhukov** [comment]
I had to rewrite the commit history because this branch had like 20 commits and merges, originally targeted master and was eventually bricked.

I feel like we have to split this PR into 2: one that upgrades dependencies and makes everything work with webpack 4, and one with the new blocks and reworked tests. While the 1st part can be safely merged (if tests pass), the 2nd part is still under consideration and API was not really discussed.

**Comment by vlad-zhukov** [comment]
@dmitmel You've reverted the branch history but the branch is inconsistent with the v2. Git (and GitHub) doesn't display the correct difference in this case (it's very unintuitive but that's how git works) unless you try to merge it into v2. If you try weird changes happen, for example `packages/eslint/package-lock.json` was added (!?).

I wanted to prevent this, that's why I squashed all commits into one.

**Comment by dmitmel** [comment]
@vlad-zhukov Can this PR be merged now?

**Comment by zcei** [comment]
@dmitmel no, I think you've reintroduced the inconsistencies again? Also we just merged the changes for moving Elm out, which conflicts with a few of your changes. Those need to be resolved first.

**Comment by dmitmel** [comment]
> I think you've reintroduced the inconsistencies again

What do you mean by "inconsistencies"?

> Also we just merged the changes for moving Elm out

I'll try to merge those later today.

**Comment by zcei** [comment]
> What do you mean by "inconsistencies"?

What Vlad described here https://github.com/andywer/webpack-blocks/pull/261#issuecomment-384062765, no?

File: packages/webpack/__tests__/end-to-end.test.js [hunk 4]
```diff
@@ -28,7 +28,7 @@ test('building the babel/postcss/extract-text project works', async (t) => {
 
   // Check if bundle contains injected process.env.TEST
   const bundleContents = await fs.readFile(path.join(buildPath, 'bundle.js'), { encoding: 'utf8' })
-  t.true(bundleContents.indexOf('module.exports = "This is the injected process.env.TEST!"') > -1)
+  t.true(bundleContents.indexOf('.exports="This is the injected process.env.TEST!"') > -1)
```
Comment (vlad-zhukov) [inline packages/webpack/__tests__/end-to-end.test.js]: This is because the code is minified by default. Changed it back in #274.

**Comment by vlad-zhukov** [comment]
I've split new blocks into #274.

This is ready for a final review! @andywer @zcei @jvanbruegge

File: packages/uglify/README.md [hunk 5]
```diff
@@ -8,16 +8,15 @@ This block provides UglifyJS webpack plugin configuration for [webpack-blocks](h
 
 Based on [uglifyjs-webpack-plugin](https://github.com/webpack-contrib/uglifyjs-webpack-plugin) (not `webpack.optimize.UglifyJsPlugin`) which uses UglifyJS v3 (uglify-es) that supports ECMAScript 2015.
 
+**This block will only work in the `production` mode.**
```
Comment (andywer) [inline packages/uglify/README.md]: Idea: Let’s turn `mode` into a link to the `setMode` docs.

Comment (vlad-zhukov) [reply packages/uglify/README.md]: In the other PR.

**Comment by vlad-zhukov** [comment]
Updated changelogs of all blocks.

@andywer It's all fine, I've squashed and rebased all commits to fix this problem.

**Comment by vlad-zhukov** [comment]
Rebased. Due to dependency changes this PR blocks lots of new ones (2 have been already opened and at least 2 more are also ready). 😅

**Comment by andywer** [comment]
Looks fine, no one's complaining. Let's merge it 😉

**Comment by dmitmel** [comment]
@andywer Let's publish a pre-relase version with webpack 4 support

**Pull request merged**